back/src/domains/marketing/use-cases/UpdateMarketingEstablishmentContactsList.ts

```diff
@@ -0,0 +1,268 @@
+            timeGateway,
+            establishment,
+            validatedConventions,
+          )
+        : onMissingEstablishment(
+            uow,
+            timeGateway,
+            establishmentMarketingGateway,
+            validatedConventions,
+            siret,
+          );
+    },
+  );
+
+const onMissingEstablishment = async (
+  uow: UnitOfWork,
+  timeGateway: TimeGateway,
+  establishmentMarketingGateway: EstablishmentMarketingGateway,
+  validatedConventions: ConventionDto[],
+  siret: SiretDto,
+): Promise<void> => {
+  const firstValidatedConvention = validatedConventions.at(0);
+  const lastValidatedConvention = validatedConventions.at(-1);
+  if (!lastValidatedConvention) {
+    return deleteMarketingContactEntity(
+      siret,
+      uow,
+      establishmentMarketingGateway,
+    );
+  }
+
+  const marketingContact: MarketingContact = {
+    createdAt: timeGateway.now(),
+    email:
+      lastValidatedConvention.signatories.establishmentRepresentative.email,
+    firstName:
+      lastValidatedConvention.signatories.establishmentRepresentative.firstName,
+    lastName:
+      lastValidatedConvention.signatories.establishmentRepresentative.lastName,
+  };
+
+  await saveMarketingContactEntity(
+    uow,
+    lastValidatedConvention.siret,
+    marketingContact,
+  );
+
+  return establishmentMarketingGateway.save({
+    isRegistered: false,
+    email: marketingContact.email,
+    firstName: marketingContact.firstName,
+    lastName: marketingContact.lastName,
+    conventions: makeConventionInfos(
+      validatedConventions,
+      firstValidatedConvention,
+      lastValidatedConvention,
+    ),
+    hasIcAccount: false,
+    numberEmployeesRange:
+      lastValidatedConvention.establishmentNumberEmployeesRange,
+    siret: lastValidatedConvention.siret,
+  });
+};
+
+const onEstablishment = async (
+  uow: UnitOfWork,
+  marketingGateway: EstablishmentMarketingGateway,
+  timeGateway: TimeGateway,
+  establishment: EstablishmentAggregate,
+  validatedConventions: ConventionDto[],
+): Promise<void> => {
+  const firstLocation = establishment.establishment.locations.at(0);
+  if (!firstLocation) throw new Error("Establishement has no location.");
+
+  const marketingContact: MarketingContact = {
+    createdAt: timeGateway.now(),
+    email: establishment.contact.email,
+    firstName: establishment.contact.firstName,
+    lastName: establishment.contact.lastName,
+  };
+
+  await saveMarketingContactEntity(
+    uow,
+    establishment.establishment.siret,
+    marketingContact,
+  );
+
+  const user = await uow.userRepository.findByEmail(marketingContact.email);
+
+  const hasIcAccount = !!user?.externalId;
+
+  const firstValidatedConvention = validatedConventions.at(0);
+  const lastValidatedConvention = validatedConventions.at(-1);
+
+  const discussions = await uow.discussionRepository.getDiscussions(
+    { sirets: [establishment.establishment.siret] },
+    500,
+  );
```

Comment:
pas facile de savoir ce que c'est que ce 500.

On peut se faire un objet ici aussi pour clarifier :

```suggestion
  const discussions = await uow.discussionRepository.getDiscussions(
    { 
      filters: { sirets: [establishment.establishment.siret] },
      limit: 500,
    }
  );
```